pkg/tokencache/repository/keyring.go

```diff
@@ -0,0 +1,57 @@
+package repository
+
+import (
+	"encoding/json"
+	"fmt"
+
+	oidc "github.com/int128/kubelogin/pkg/oidc"
+	tokencache "github.com/int128/kubelogin/pkg/tokencache"
+	"github.com/zalando/go-keyring"
+)
+
+// Keyring provides access to the token cache using OS keyring.
+// Keys of a token cache is sha256 digest of the issuer, zero-character and client ID.
+type Keyring struct{}
```

Comment:
Should it instead always try using the OS keyring, and only use the file based approach as a fallback? Such as:

- First try read/write from keyring
- skip if `--no-keyring` flag is provided
- if not ["unsupported platform" error](https://github.com/zalando/go-keyring/blob/v0.2.3/keyring_fallback.go#L9), then just return error, aborting the command.
- if no secret found in keyring, continue with file based implementation

That would preserve a user's old tokens, while letting them slowly transition into OS keyring based cache.

It would be more intrusive, but maybe changing the default implementation is a good idea here?

Reply:
I just changed my implementation to match above.